metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -2349,14 +2350,25 @@ class MetalsLspService(
                 importBuild,
               )
           }
-        case None =>
+        case Some(buildTool: BuildServerProvider) =>
+          buildTool.digest(workspace) match {
+            case None =>
+              scribe.warn(s"Skipping generating bsp files, no checksum.")
+              Future.successful(BuildChange.None)
+            case Some(digest) if !buildTools.isBsp =>
+              generateBspConfig()
+              indexer.reloadWorkspaceAndIndex(forceImport, buildTool, digest)
```

Comment:
`reloadWorkspaceAndIndex` takes 4 args now and make sure to run `sbt scalafixAll` :)